### PR TITLE
Allow negation

### PR DESCRIPTION
**Pull request by dim**
It's a bigger change than it may seem, will explain inline

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,15 +17,15 @@ class Comment < ActiveRecord::Base
   belongs_to :post
 
   filterable_by :post_id, :user_id
-  filterable_by :post_author_id do |scope, value|
-    scope.joins(:posts).where(:'posts.author_id' => value)
+  filterable_by :post_author_id do |value|
```
Comment (dim) [inline README.md:20]: so the big change is that you don't now need `scope` as a block argument, but the old method is still supported (although deprecated)

File: README.md [hunk 2]
```diff
@@ -39,6 +39,12 @@ Simple use cases:
 Comment.filter_by({ 'post_id' => '1' })
 # => WHERE post_id = 1
 
+Comment.filter_by({ 'post_id' => ['1', '2'] })
+# => WHERE post_id IN (1, 2)
+
+Comment.filter_by({ 'post_id_not' => '3' })
```
Comment (dim) [inline README.md:45]: the other change is that you now can append `_not` and exclude something

Comment (andyborn) [reply README.md:45]: will we still be able to handle when value is nil cases?

**Review by mxmCherry — APPROVED** [review]

**Comment by dim** [comment]
@jpmedforth @andyborn ?

**Review by andyborn — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (dim) [reply README.md:20]: value cannot be nil, these blocks are only run `unless value.bank?`

File: lib/filterable_by.rb [hunk 3]
```diff
@@ -13,6 +13,28 @@ def self.normalize(value)
       end
     end
 
+    def self.merge(scope, unscoped, hash, name, **opts, &block)
+      key = name
+      positive = normalize(hash[key]) if hash.key?(key)
+      if positive.present?
```
Comment (dim) [inline lib/filterable_by.rb:19]: @andyborn here we only apply block/limit scope if `params['post_author_id']` is present

File: lib/filterable_by.rb [hunk 4]
```diff
@@ -13,6 +13,28 @@ def self.normalize(value)
...
+        sub = block.arity == 2 ? yield(unscoped, positive, **opts) : yield(positive, **opts)
+        return nil unless sub
+
+        scope = scope.merge(sub)
+      end
+
+      key = "#{name}_not"
+      negative = normalize(hash[key]) if hash.key?(key)
+      if negative.present?
```
Comment (dim) [inline lib/filterable_by.rb:28]: same with `params['post_author_id_not']`

**Comment by dim** [comment]
Released v0.6.0

**Pull request merged**